docs/source/getting_started/UserSessionConfig.rst

```diff
@@ -0,0 +1,67 @@
+Automatic Loading of Query Providers and Components
+===================================================
+
+The ``mp_user_session.py`` module is designed to load and initialize query providers and other
+components based on configuration provided in a YAML file.
+
+This allows you to load multiple providers and components in a single step avoiding having to
+write a lot of repetitive code in your notebooks.
+
+The user is expected to supply the path to the YAML file to the ``load_user_config`` function.
+Each key in the ``QueryProviders`` and ``Components`` sections of the YAML file will be the name
+of the component variable in the local namespace.
+
+Example YAML Configuration
+--------------------------
+
+Here is an example of a YAML configuration file that defines query providers and components:
+
+.. code-block:: yaml
+
+    QueryProviders:
+        qry_prov_sent:
+            DataEnvironment: MSSentinel
+            InitArgs:
+                debug: True
+            Connect: True
+            ConnectArgs:
+                workspace: CyberSecuritySoc
+                auth_methods: ['cli', 'device_code']
+        qry_prov_md:
+            DataEnvironment: M365D
+        qry_kusto_mde:
+            DataEnvironment: Kusto
+            Connect: True
+            ConnectArgs:
+                cluster: MDE-Scrubbed
+    Components:
+        mssentinel:
+            Module: msticpy.context.azure
+            Class: MicrosoftSentinel
+            InitArgs:
+            Connect: True
```

Comment:
I think this needs to be indented one level

Reply:
Not sure if this is what you meant but I did have a misaligned space - QueryProviders was 1 char less indented than Components. Enough to cause pyyaml to fail!